### PR TITLE
Build mafia using GHC 8.0

### PR DESCRIPTION
**Pull request by jacobstanley**
The tests for this depend on x being upgraded to ghc 8.0

/cc @tmcgilchrist @thumphries @erikd-ambiata

### CODE REVIEW

File: main/mafia.hs [hunk 1]
```diff
@@ -56,6 +57,8 @@ main = do
   dispatch parser >>= \case
     VersionCommand ->
       putStrLn buildInfoVersion >> exitSuccess
+    DependencyCommand ->
+      traverse putStrLn dependencyInfo >> exitSuccess
```
Comment (tmcgilchrist) [inline main/mafia.hs:61]: Do you need to add `gen` to the `hs-source-dirs` for this executable?

Comment (tmcgilchrist) [inline main/mafia.hs:61]: Looks like it's already there, never mind.

**Comment by tmcgilchrist** [comment]
Once 8.0.1 build is good 🌽

**Comment by thumphries** [comment]
lgtm 🍧

**Pull request merged**